src/commands/transaction/mosaicglobalrestriction.ts

```diff
@@ -0,0 +1,123 @@
+    })
+    mosaicId: string;
+
+    @option({
+        flag: 'r',
+        default: '0',
+        description: '(Optional) Identifier of the mosaic providing the restriction key.',
+    })
+    referenceMosaicId: string;
+
+    @option({
+        flag: 'k',
+        description: 'Restriction key relative to the reference mosaic identifier.',
+        validator: new MosaicRestrictionKeyValidator(),
+    })
+    restrictionKey: string;
+
+    @option({
+        flag: 'v',
+        default: '0',
+        description: '(Optional) Previous restriction value.',
+        validator: new NumericStringValidator(),
+    })
+    previousRestrictionValue: string;
+
+    @option({
+        flag: 't',
+        default: 'NONE',
+        description: '(Optional) Previous restriction type. (NONE: no restriction, EQ: equal, NE: not equal, LT: less than,' +
+            'LE: less than or equal, GT: greater than, GE: greater than or equal)',
+        validator: new MosaicRestrictionTypeValidator(),
+    })
+    previousRestrictionType: string;
+
+    @option({
+        flag: 'V',
+        description: 'New restriction value.',
+        validator: new NumericStringValidator(),
+    })
+    newRestrictionValue: string;
+
+    @option({
+        flag: 'T',
+        description: 'New restriction type.',
+        validator: new MosaicRestrictionTypeValidator(),
+    })
+    newRestrictionType: string;
+}
+
+@command({
+    description: 'Set a global restriction to a mosaic',
+})
+export default class extends AnnounceTransactionsCommand {
+    constructor() {
+        super();
+    }
+    @metadata
+    execute(options: CommandOptions) {
+        const profile = this.getProfile(options);
+        const account = profile.decrypt(options);
+        const mosaicId = new MosaicIdAliasResolver().resolve(options);
+        const newRestrictionType = new RestrictionTypeResolver().resolve(options);
+        const restrictionKey = new RestrictionKeyResolver().resolve(options);
+        const newRestrictionValue = new RestrictionValueResolver().resolve(options);
+        const maxFee = new MaxFeeResolver().resolve(options);
+
+        const previousRestrictionType = Number(MosaicRestrictionType[options.previousRestrictionType as any]) as MosaicRestrictionType;
+        const previousRestrictionValue = UInt64.fromNumericString(options.previousRestrictionValue);
+        const referenceMosaicId = options.referenceMosaicId === '0' ? undefined : new MosaicId(options.referenceMosaicId);
+
+        const transaction = MosaicGlobalRestrictionTransaction.create(
+            Deadline.create(),
+            mosaicId,
+            restrictionKey,
+            previousRestrictionValue,
+            previousRestrictionType,
+            newRestrictionValue,
+            newRestrictionType,
+            profile.networkType,
+            referenceMosaicId,
+            maxFee,
+        );
+
+        const networkGenerationHash = profile.networkGenerationHash;
```

Comment:
Add support for the recently added ``shouldAnnounce`` and ``sync`` options:

https://github.com/nemtech/nem2-cli/blob/master/src/commands/transaction/accountlink.ts#L69-L75